src/components/run-button-container.js

```diff
@@ -221,289 +192,278 @@ export function RunButtonContainer({ studyUuid, currentNode, disabled }) {
         loadFlowStatus,
     ]);
 
-    const ACTION_ON_RUNNABLES = {
-        text: intl.formatMessage({ id: 'StopComputation' }),
-        action: (action) => {
-            let type;
-            switch (action) {
-                case runnable[ComputingType.SECURITY_ANALYSIS]:
-                    type = ComputingType.SECURITY_ANALYSIS;
-                    stopSecurityAnalysis(studyUuid, currentNode?.id);
-                    break;
-                case runnable[ComputingType.SENSITIVITY_ANALYSIS]:
-                    type = ComputingType.SENSITIVITY_ANALYSIS;
-                    stopSensitivityAnalysis(studyUuid, currentNode?.id);
-                    break;
-                case runnable[ComputingType.ALL_BUSES_SHORTCIRCUIT_ANALYSIS]:
-                    type = ComputingType.ALL_BUSES_SHORTCIRCUIT_ANALYSIS;
-                    stopShortCircuitAnalysis(studyUuid, currentNode?.id);
-                    break;
-                case runnable[ComputingType.DYNAMIC_SIMULATION]:
-                    type = ComputingType.DYNAMIC_SIMULATION;
-                    stopDynamicSimulation(studyUuid, currentNode?.id);
-                    break;
-                case runnable[ComputingType.VOLTAGE_INIT]:
-                    type = ComputingType.VOLTAGE_INIT;
-                    stopVoltageInit(studyUuid, currentNode?.id);
-                    break;
-                case runnable[ComputingType.LOADFLOW]:
-                    type = ComputingType.LOADFLOW;
-                    stopLoadFlow(studyUuid, currentNode?.id);
-                    break;
-                default:
-                    return;
-            }
-
-            dispatch(setComputingStatus(type, RunningStatus.IDLE));
-            setComputationStopped(true);
+    const startComputationAsync = useCallback(
+        (computingType, fnBefore, fnStart, fnThen, fnCatch, errorHeaderId) => {
+            dispatch(setComputingStatus(computingType, RunningStatus.RUNNING));
+            fnStart.then(fnThen).catch((error) => {
```

Comment:
Since fnStart is an arrow function, shouldn't it be :
```suggestion
            fnStart().then(fnThen).catch((error) => {
```

Reply:
yes, thank